packages/autoloader/src/AutoloadGenerator.php

```diff
@@ -73,56 +80,26 @@ public function dump(
-		$this->io->writeError( '<info>Generated ' . $targetDir . '/jetpack_autoload_classmap.php</info>', true );
-
-		file_put_contents( $targetDir . '/jetpack_autoload_filemap.php', $this->getAutoloadFilesPackagesFile( $fileMap ) );
-		$this->io->writeError( '<info>Generated ' . $targetDir . '/jetpack_autoload_filemap.php</info>', true );
-
-		file_put_contents( $vendorPath . '/autoload_packages.php', $this->getAutoloadPackageFile( 'autoload.php', $suffix ) );
-		$this->io->writeError( '<info>Generated ' . $vendorPath . '/autoload_packages.php</info>', true );
-
-		$jetpackAutoloaderDir = $vendorPath . '/jetpack-autoloader';
-		$filesystem->ensureDirectoryExists( $jetpackAutoloaderDir );
-		file_put_contents( $jetpackAutoloaderDir . '/autoload_functions.php', $this->getAutoloadPackageFile( 'functions.php', $suffix ) );
-		$this->io->writeError( '<info>Generated ' . $jetpackAutoloaderDir . '/jetpack-autoloader/autoload_functions.php</info>', true );
-
-		file_put_contents( $vendorPath . '/class-autoloader-handler.php', $this->getAutoloadPackageFile( 'class-autoloader-handler.php', $suffix ) );
-		$this->io->writeError( '<info>Generated ' . $vendorPath . '/class-autoloader-handler.php</info>', true );
+		// Convert the autoloads into a format that the manifest generator can consume more easily.
+		$basePath           = $this->filesystem->normalizePath( realpath( getcwd() ) );
+		$vendorPath         = $this->filesystem->normalizePath( realpath( $config->get( 'vendor-dir' ) ) );
+		$processedAutoloads = $this->processAutoloads( $autoloads, $scanPsrPackages, $vendorPath, $basePath );
+		unset( $packageMap, $autoloads );
 
-		file_put_contents( $vendorPath . '/class-classes-handler.php', $this->getAutoloadPackageFile( 'class-classes-handler.php', $suffix ) );
-		$this->io->writeError( '<info>Generated ' . $vendorPath . '/class-classes-handler.php</info>', true );
+		// Make sure none of the legacy files remain that can lead to problems with the autoloader.
+		$this->removeLegacyFiles( $vendorPath );
 
-		file_put_contents( $vendorPath . '/class-files-handler.php', $this->getAutoloadPackageFile( 'class-files-handler.php', $suffix ) );
-		$this->io->writeError( '<info>Generated ' . $vendorPath . '/class-files-handler.php</info>', true );
-
-		file_put_contents( $vendorPath . '/class-plugins-handler.php', $this->getAutoloadPackageFile( 'class-plugins-handler.php', $suffix ) );
-		$this->io->writeError( '<info>Generated ' . $vendorPath . '/class-plugins-handler.php</info>', true );
-
-		file_put_contents( $vendorPath . '/class-version-selector.php', $this->getAutoloadPackageFile( 'class-version-selector.php', $suffix ) );
-		$this->io->writeError( '<info>Generated ' . $vendorPath . '/class-version-selector.php</info>', true );
+		// Write all of the files now that we're done.
+		$this->writeAutoloaderFiles( $vendorPath . '/jetpack-autoloader/', $suffix );
+		$this->writeManifests( $vendorPath . '/' . $targetDir, $processedAutoloads );
 	}
```

Comment:
Do you think there would be any value in displaying a warning when an unoptimized autoloader is generated? Something like this:

```
if ( ! $scanPsrPackages ) {
    $this->io->writeError( "<warning>Generating an unoptimized autoloader. If this is a production build, consider using the -o flag to generate an optimized autoloader.</warning>" );
}
```

Reply:
I was actually planning to add this today as I'd like to avoid introducing silent problems for those who update to this next version of the autoloader. I know for instance that in our build scripts we use `composer install` and so we're going to need to address that.